### PR TITLE
test: migrate delete test from tap to node test runner

### PR DESCRIPTION
**Pull request by Kalpana98**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Kalpana98** [comment]
#5555

**Comment by Kalpana98** [comment]
@mcollina please review.

**Review by dancastillo — COMMENTED** [review]
Can you convert `t.assert.equal` to `t.assert.strictEqual` and `t.assert.deepEqual` to `t.assert.deepStrictEqual`

**Comment by Kalpana98** [comment]
> Can you convert `t.assert.equal` to `t.assert.strictEqual` and `t.assert.deepEqual` to `t.assert.deepStrictEqual`

@dancastillo done :)

File: test/delete.test.js [hunk 1]
```diff
@@ -151,60 +154,63 @@ test('body - delete', t => {
...
   } catch (e) {
-    t.fail()
+    t.assert.fail()
   }
 })
 
 fastify.listen({ port: 0 }, err => {
-  t.error(err)
-  t.teardown(() => { fastify.close() })
+  ifError(err)
```
Comment (gurgunday) [inline test/delete.test.js:164]: `ifError` doesn't exist in `t`?

```suggestion
  t.assert.ifError(err)
```

Comment (Kalpana98) [reply test/delete.test.js:164]: @gurgunday `t` provides the `t.assert.ifError(err)` utility, scoped only to individual test blocks. For global use, such a utility does not exist in `t`. Therefore, it needs to be imported from the `assert`.

File: test/delete.test.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 'use strict'
 
-const t = require('tap')
+const { ifError } = require('node:assert')
+const t = require('node:test')
 const test = t.test
```
Comment (jsumners) [inline test/delete.test.js:5]:
```suggestion
const test = require('node:test')
```

Comment (Kalpana98) [reply test/delete.test.js]: updated @jsumners

File: test/delete.test.js (hunk 1 above)
Comment (dancastillo) [reply test/delete.test.js:163]: wondering if we need this we could also do something like this. @gurgunday wdyt?
```js
fastify.listen({ port: 0 }, err => {
  if (err) throw new Error('Fastify failed to start')
  .....
})
```

Comment (Kalpana98) [reply test/delete.test.js:163]: @dancastillo @jsumners any suggestions for me?

Comment (jsumners) [reply test/delete.test.js:163]:
```js
const assert = require('node:assert')

fastify.listen(0, (error) => {
	assert.ifError(error)
})
```

Comment (Kalpana98) [reply test/delete.test.js]: @jsumners done

**Review by jsumners — APPROVED** [review]

**Pull request merged**